### PR TITLE
Changes to report

### PR DESCRIPTION
**Pull request by hbandukw**
1) Revamped the Methods section
2) I hid the code cells in the jupyter book HTML
3) Added figure captions but in a hacky way (i.e. in a new cell) because jupyter book code was not working. 

Things to do:
1) We need to add F1 or accuracy scores in the text where appropriate. 
2) Results, discussion and limitations are still incomplete.

### CODE REVIEW

**Comment by sean-m-mckay** [comment]
Can you remove the build files? (I'll update the gitignore in my next PR so that's done automatically)

File: report/wine_classification_report.ipynb [hunk 1]
```diff
@@ -144,7 +156,7 @@
...
     "\n",
-    "With this project, we intend to showcase that this methodology has potential of streamlining wine identification processes for the benefit of the industry. Since this model is intended as a proof-of-concept, it can be improved significantly by considering other important physiochemical properties and modern techniques for feature selection. The model can also be refined with further testing with larger and more complex datasets."
+    "With this project, we intend to showcase that this methodology has potential of streamlining wine identification processes for the benefit of the industry. Since this model is intended as a proof-of-concept, it can be improved significantly by considering other important physiochemical features and modern techniques for feature selection. The model can also be refined with a larger and more complex datasets."
```
Comment (sean-m-mckay) [inline report/wine_classification_report.ipynb:159]: Minor, "a larger .. datasets" can remove the "a"

File: report/wine_classification_report.ipynb [hunk 2]
```diff
@@ -195,28 +207,58 @@
...
-    "# Results and Discussion"
+    "## Methods "
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "For our data analysis, we first split the data into train and test sets with an equal distribution of target classes in each set to ensure the model generalizes well. The train-test split was done before any further data analysis and scaling to avoid information leakage. All of the features in the dataset are numerical, so we applied the standard scaler to all of them to ensure they take on the same range of values.\n",
+    "We first separated our dataset into random train and test sets using sklearn's `train_test_split`. The split between train and test sets was 70:30% respectively. We ensures that there would be an equal distribution of target classes across the two sets with the `stratify` argument.  Additionally, a `random_state` was defined to ensure the reproducibility of our results. \n",
```
Comment (sean-m-mckay) [inline report/wine_classification_report.ipynb:224]: Minor: "ensures" -> "ensured"

**Comment by hbandukw** [comment]
> Can you remove the build files? (I'll update the gitignore in my next PR so that's done automatically)

Can you describe how you are rendering the HTML then? Are the build files going to be compiled automatically within the github pages procedure?

File: report/wine_classification_report.ipynb [hunk 3]
```diff
@@ -225,28 +267,134 @@
...
+   "metadata": {},
+   "source": [
+    "Next, we built our LR model with sklearn's `LogisticRegression`. Logistic regression was our linear model of choice because it is simple, easy to implement/interpret and can also extend to multiple classes (as is the case with our dataset). "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To find the optimal value of the hyperparameter `C` for our LR model that maximized the accuracy score, we performed a 10-fold cross-validated grid search with sklearn's `GridSearchCV`.\n",
```
Comment (sean-m-mckay) [inline report/wine_classification_report.ipynb:309]: Minor: We could explain we used GridSearch over RandomSearch since there's only one parameter to Optimize

File: report/wine_classification_report.ipynb [hunk 4]
```diff
@@ -225,28 +267,134 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "  Metrics such as precision, recall, F1-score are good choices for imblanced class. According to our baseline model, the accuracy is of 40.33%, indicates that the most prevalent class occurs at a rate of 40%. This suggests a class imbalance (as we have three classes) "
```
Comment (sean-m-mckay) [inline report/wine_classification_report.ipynb:397]: This is not really a limitation, something we're already accounting for in the model - limitations we can mention are maybe we're not sure how the model generalizes to Cultivars outside of the 3 we're classifying in the model

**Comment by sean-m-mckay** [comment]
> > Can you remove the build files? (I'll update the gitignore in my next PR so that's done automatically)
> 
> Can you describe how you are rendering the HTML then? Are the build files going to be generated automatically from the notebook? Can we leave them in while I work on the report because I am unsure about the process you implemented?

Yes, the build files are generated automatically anytime a new commit to main is made, and pushed to the github_pages branch

The build process that does that is here: https://github.com/UBC-MDS/wine-origin-prediction/actions/workflows/deploy-book.yml

File: report/wine_classification_report.ipynb (hunk 4 above)
Comment (sean-m-mckay) [reply report/wine_classification_report.ipynb:397]: We'll address in subsequent PR, gonna just merge this for now to get rid of conflicts

**Review by sean-m-mckay — APPROVED** [review]
